projects/libs/core-library-typescript/src/gossip/gossip.ts

```diff
@@ -51,37 +52,35 @@ export class Gossip {
     });
 
     ws.on("message", (message: Buffer) => {
-      let offset = 0;
-
-      function readString(buf: Buffer): string {
-        const len: number = buf.readUInt8(offset);
-        const str: string = buf.toString("utf8", offset + 1, offset + 1 + len);
-        offset += 1 + len;
-        return str;
-      }
-
-      const sender = readString(message);
+      const rawMessage = JSON.parse(message.toString());
+      const { sender, hello, recipient, broadcastGroup, objectType, buffer, signature } = rawMessage;
 
       if (this.localAddress == sender) {
         logger.info("Connected to myself. Disconnecting");
         ws.close();
         return;
       }
 
-      if (offset === message.length) {
+      if (hello) {
```

Comment:
![image](https://user-images.githubusercontent.com/2841614/38289044-2c627566-380f-11e8-92ef-3f13d833513f.png)

Reply:
![neil-degrasse-tyson-reveals-the-biggest-misconceptions-people-have-about-the-universe](https://user-images.githubusercontent.com/614125/38311560-2883f4ec-3828-11e8-90fc-f9838a4430d6.jpg)